### PR TITLE
Fix DebuggerDisplay for serializer types

### LINKED ISSUE #65960: [DebuggerDisplay] for JsonTypeInfo does not render
### Description

While debugging some code using the JSON source generator in Visual Studio 2022 I found that the debugger display for a `JsonSerializerContext` class is broken.

image

Instead of showing the intended display text, instead the IDE shows a message similar to `ConverterStrategy.error CS0119: 'ConverterStrategy' is a type, which is not valid in the given context, "Boolean"`.

Following the code through, it looks like there's an issue with the value passed to the constructor of the `[DebuggerDisplay]` attribute on `JsonTypeInfo` here:

https://github.com/dotnet/runtime/blob/c41f452cc96a2f2e73c71610d423df3e9c133fef/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs#L17

### Reproduction Steps

View a derived instance of `` using the JSON source generator in an IDE with Intellisense, such as Visual Studio 2022.

```csharp
using System.Text.Json.Serialization;

var context = new CustomJsonSerializerContext();

// Put a breakpoint on the line below and inspect context in the debugger
Console.ReadLine();

[JsonSerializable(typeof(bool))]
internal sealed partial class CustomJsonSerializerContext : JsonSerializerContext
{
}
```

### Expected behavior

A compiler error is not shown in the rendered debugger display for the class' properties.

### Actual behavior

The properties all show a compiler error similar to the below.

```
ConverterStrategy.error CS0119: 'ConverterStrategy' is a type, which is not valid in the given context, "Boolean".
```

### Regression?

No, this functionality was new in .NET 6.

### Known Workarounds

None.

### Configuration

```
> dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.200
 Commit:    4c30de7899

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.200\
```

### Other information

_No response_

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Fix #65960.

### CODE REVIEW

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs [hunk 1]
```diff
@@ -120,5 +120,8 @@ public JsonConverter InitializeReEntry(Type type, JsonSerializerOptions options)
 
             return PolymorphicJsonPropertyInfo.ConverterBase;
         }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private string DebuggerDisplay => $"ConverterStrategy.{JsonTypeInfo?.PropertyInfoForTypeInfo.ConverterStrategy}, {JsonTypeInfo?.Type.Name}";
```
Comment (layomia) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs:125]: In some cases we read from `JsonPropertyInfo.ConverterStrategy` while in others, we read from `JsonPropertyInfo.ConverterBase.ConverterStrategy`. Could this be made consistent?

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs [hunk 2]
```diff
@@ -355,5 +355,8 @@ private void SetConstructorArgumentState()
                 }
             }
         }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private string DebuggerDisplay => $"Path:{JsonPath()} Current: ConverterStrategy.{Current.JsonTypeInfo?.PropertyInfoForTypeInfo.ConverterStrategy}, {Current.JsonTypeInfo?.Type.Name}";
```
Comment (stephentoub) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs:360]: Nit: The use of "=", ":", and ": " as key/value separators across the debugger displays looks inconsistent.

Comment (stephentoub) [reply src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs:360]: Similarly, some use ", " between pairs while others use " "

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs [hunk 3]
```diff
@@ -519,5 +519,8 @@ internal JsonTypeInfo JsonTypeInfo
         /// Default value used for parameterized ctor invocation.
         /// </summary>
         internal abstract object? DefaultValue { get; }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
```
Comment (stephentoub) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs:523]: Why use this attribute?

**Review by layomia — APPROVED** [review]

**Comment by TonyValenti** [comment]
@eiriktsarpalis Can I make a suggestion?

In my own code, I create a constant that contains the debuggerdisplay,nq value and reference the constant everywhere.  This keeps me from having to have the magic string everywhere. 

Also, as a general rule, I've found it better to be a function rather than a property because sometimes it will wind up being serialized in weird places if it is a property. 

Anyway, just a few thoughts. Do with them what you will.

**Pull request merged**